### PR TITLE
Do not show 'all' limit selector option for VisitorLog & default to 25 if filter_limit=-1 used

### LINKED ISSUE #12789: Visitor log: do not allow to show "all" rows as it can time out
In Visitor Log, Ecommerce log, Media log, etc. it is currently possible to select `all` in the Rows selector, meaning that all visits will be displayed in the log. 

The idea was to allow users to export "All" the visitor log data for a day. But since Matomo 3.4.0, a new export feature in https://github.com/matomo-org/matomo/pull/11958 is available, which lets user select the number of rows to export. 

Therefore, it is not needed anymore to offer "All" in the rows selector for the purpose of exporting all data. 

## The problem

For many websites, selecting "All" will actually cause Matomo to time out with the following error:

```
Error: {"type":1,"message":"Maximum execution time of 240 seconds Exceeded", 
"file":"core\/EventDispatcher.php","line":105}
at index.php?forceView=1&viewDataTable=VisitorLog&module=Live&action=getLastVisitsDetails&small=1&idSite=21&period=day&date=2018-04-24&showtitle=1&random=2790

```

 
## Suggested steps

* Remove `all` from the rows selector from the Visitor Log, Ecommerce log, Media log, etc.
* In the update, also delete all users' viewdatatable visualisations settings for Visitor log(s) where the setting includes "showing all rows". This way, any user who has already selected "All" will be reset to showing 25 rows or so by default.

As a result, we would expect a lot less timeouts triggered in Matomo.

### PR DESCRIPTION
**Pull request by diosmosis**
Changes:
* added new `disable_all_rows_filter_limit` visualization property that can be used to disable the 'all' option from the limit selector.
* use in VisitorLog to disable the 'all' option.
* if -1 filter_limit found when trying to load VisitorLog, change it to another value.
* change default used in this case from 20 => General.datatable_default_limit.

Fixes #12789

### CODE REVIEW

File: plugins/Live/Visualizations/VisitorLog.php [hunk 1]
```diff
@@ -43,8 +43,10 @@ public function beforeLoadDataTable()
             'filter_sort_order',
         ));
 
-        if (!is_numeric($this->requestConfig->filter_limit)) {
-            $this->requestConfig->filter_limit = 20;
+        if (!is_numeric($this->requestConfig->filter_limit)
+            || $this->requestConfig->filter_limit == -1 // 'all' is not supported for this visualization
+        ) {
+            $this->requestConfig->filter_limit = 25;
```
Comment (mattab) [inline plugins/Live/Visualizations/VisitorLog.php]: here let's use `General.datatable_default_limit`

**Review by mattab — CHANGES_REQUESTED** [review]
Looks good, just a minor feedback

**Comment by diosmosis** [comment]
Oh whoops, didn't realize that would happen to the UI test viewer. I'll create a new branch to view tests results from

**Comment by diosmosis** [comment]
Created branch to view build here: https://travis-ci.org/matomo-org/matomo/builds/373122394

**Comment by diosmosis** [comment]
Some UI test failures, should be fixed on merge since they will likely conflict w/ other PRs.

**Comment by diosmosis** [comment]
@mattab can you approve if this is good to go now? will update the UI tests & merge at that point

**Review by mattab — APPROVED** [review]

**Pull request merged**